tests/hardware/awg_new_driver_base_tests.py

```diff
@@ -0,0 +1,270 @@
+from typing import Callable, Iterable, Optional, Set, Tuple
+import unittest
+import warnings
+
+from qupulse import ChannelID
+from qupulse._program._loop import Loop
+from qupulse.hardware.awgs_new_driver.base import AWGDevice, AWGChannel, AWGChannelTuple, AWGMarkerChannel
+from qupulse.hardware.awgs_new_driver.features import ChannelSynchronization, ProgramManagement, VoltageRange, \
+    AmplitudeOffsetHandling
+from qupulse.utils.types import Collection
+
+warnings.simplefilter("ignore", UserWarning)
```

Comment:
Please use a context manger to catch warnings when necessary. Otherwise this silences warnings we want occure during testing.

Reply:
I removed it